plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2273,7 +2273,9 @@ private TableSnapshot getSnapshot(ConnectorSession session, DeltaLakeTableHandle
     private TableSnapshot getSnapshot(ConnectorSession session, SchemaTableName schemaTableName, String tableLocation, Optional<Long> atVersion)
     {
         try {
-            return transactionLogAccess.getSnapshot(session, schemaTableName, tableLocation, atVersion);
+            synchronized (this) {
```

Comment:
what is so special about this call to `transactionLogAccess.getSnapshot` that you are synchronizing it? There is a bunch of other calls without synchroniation

Reply:
Is that always initial in query processing? Please comment.

Reply:
Also nit: maybe move the synchronization logic to `TransactionLogAccess` itslef. You can have two methods (e.g. `transactionLogAccess.getSnapshot` and `transactionLogAccess.waitAndGetSnapshot`).
The other thing is that synchronizaing whole structure for IO heavy method is not great. Consider having lock per cache key.

Reply:
will add a comment here
```
 // This method is not really called concurrently. If it were, synchronize to ensure we load table state
 // once in TransactionLogAccess. TODO move query-level (transaction-level) state out of TransactionLogAccess singleton
```

the point is -- more complex synchronization is not needed, currently it's for documentation purposes